maven-release-plugin/src/main/java/org/apache/maven/plugins/release/AbstractScmReadReleaseMojo.java

```diff
@@ -167,11 +125,11 @@ protected ReleaseDescriptorBuilder createReleaseDescriptor() {
             }
         }
 
-        // As long as Scm.getId() does not exist, read it as a property
-        descriptor.setScmId(project.getProperties().getProperty("project.scm.id"));
+        descriptor.setScmId(serverId);
 
         for (MavenProject reactorProject : session.getProjects()) {
-            if (reactorProject.getScm() != null) {
+            if (reactorProject.getOriginalModel() != null
+                    && reactorProject.getOriginalModel().getScm() != null) {
```

Comment:
This is a cause for: #1420, now `originalScmInfo` only contains project with defined scm info. 
The change is ok.